### PR TITLE
Feature/fix small issues

### PR DESCRIPTION
**Pull request by ganfra**
This PR fixes the following issues and some others I could have encountered using the app: 

 - Integration Manager: Wrong URL to review terms if URL in config contains path (#1606)
 - Regression Composer does not grow, crops out text (#1650)
 - Bug / Unwanted draft (#698)
 - All users seems to be able to see the enable encryption option in room settings (#1341)
 - Leave room only leaves the current version (#1656)

### CODE REVIEW

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/DefaultRoomDirectoryService.kt [hunk 1]
```diff
@@ -44,14 +44,6 @@ internal class DefaultRoomDirectoryService @Inject constructor(private val getPu
                 .executeBy(taskExecutor)
     }
 
-    override fun joinRoom(roomIdOrAlias: String, reason: String?, callback: MatrixCallback<Unit>): Cancelable {
-        return joinRoomTask
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/DefaultRoomDirectoryService.kt:48]: joinRoomTask can be remove from constructor then

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/membership/leaving/LeaveRoomTask.kt [hunk 2]
```diff
@@ -31,12 +40,40 @@ internal interface LeaveRoomTask : Task<LeaveRoomTask.Params, Unit> {
...
+        roomChangeMembershipStateDataSource.updateState(roomId, ChangeMembershipState.Leaving)
+        val roomCreateStateEvent = stateEventDataSource.getStateEvent(
+                roomId = roomId,
+                eventType = EventType.STATE_ROOM_CREATE,
+                stateKey = QueryStringValue.NoCondition
+        )
+        // Server is not cleaning predecessor rooms, so we also try to left them
+        val predecessorRoomId = roomCreateStateEvent?.getClearContent()?.toModel<RoomCreateContent>()?.predecessor?.roomId
+        if (predecessorRoomId != null) {
+            leaveRoom(predecessorRoomId, reason)
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/membership/leaving/LeaveRoomTask.kt:68]: If it fails for some reason, it should maybe not be fatal?

File: vector/src/main/java/im/vector/riotx/core/platform/EllipsizingTextView.kt [hunk 3]
```diff
@@ -1,421 +0,0 @@
...
-/**
- * A [android.widget.TextView] that ellipsizes more intelligently.
- * This class supports ellipsizing multiline text through setting `android:ellipsize`
- * and `android:maxLines`.
- *
- *
- * Note: [TruncateAt.MARQUEE] ellipsizing type is not supported.
- * This as to be used to get rid of the StaticLayout issue with maxLines and ellipsize causing some performance issues.
- */
-class EllipsizingTextView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyle: Int = android.R.attr.textViewStyle)
```
Comment (bmarty) [inline vector/src/main/java/im/vector/riotx/core/platform/EllipsizingTextView.kt:61]: It has been introduced to fix ellipsis, so is it ok to remove it now?

File: vector/src/main/java/im/vector/riotx/features/home/room/detail/RoomDetailFragment.kt [hunk 4]
```diff
@@ -293,6 +293,7 @@ class RoomDetailFragment @Inject constructor(
         setupJumpToBottomView()
         setupWidgetsBannerView()
 
+        roomToolbarContentView.isClickable = false
```
Comment (bmarty) [inline vector/src/main/java/im/vector/riotx/features/home/room/detail/RoomDetailFragment.kt:296]: Useless, I think next line will restore isClickable to true

File: vector/src/main/java/im/vector/riotx/features/home/room/detail/timeline/format/NoticeEventFormatter.kt [hunk 5]
```diff
@@ -95,7 +95,7 @@ class NoticeEventFormatter @Inject constructor(private val activeSessionDataSour
 
     private fun formatRoomPowerLevels(event: Event, disambiguatedDisplayName: String): CharSequence? {
         val powerLevelsContent: PowerLevelsContent = event.getClearContent().toModel() ?: return null
-        val previousPowerLevelsContent: PowerLevelsContent = event.prevContent.toModel() ?: return null
+        val previousPowerLevelsContent: PowerLevelsContent = event.resolvedPrevContent().toModel() ?: return null
```
Comment (bmarty) [inline vector/src/main/java/im/vector/riotx/features/home/room/detail/timeline/format/NoticeEventFormatter.kt:98]: I was wondering why we were not using `resolvedPrevContent()` everywhere. Not clear in the spec about which part of the Json has to be taken into account. So Ok for these change!

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/membership/leaving/LeaveRoomTask.kt (hunk 2 above)
Comment (ganfra) [reply matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/membership/leaving/LeaveRoomTask.kt:68]: I start from the predecessors,so I think it's ok

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/DefaultRoomDirectoryService.kt (hunk 1 above)
Comment (ganfra) [reply matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/DefaultRoomDirectoryService.kt:48]: Ya

File: vector/src/main/java/im/vector/riotx/core/platform/EllipsizingTextView.kt (hunk 3 above)
Comment (ganfra) [reply vector/src/main/java/im/vector/riotx/core/platform/EllipsizingTextView.kt:61]: It adds more issue than it solves

File: vector/src/main/java/im/vector/riotx/features/home/room/detail/RoomDetailFragment.kt (hunk 4 above)
Comment (ganfra) [reply vector/src/main/java/im/vector/riotx/features/home/room/detail/RoomDetailFragment.kt:296]: Yes nice catch

File: vector/src/main/java/im/vector/riotx/features/home/room/detail/timeline/format/NoticeEventFormatter.kt (hunk 5 above)
Comment (ganfra) [reply vector/src/main/java/im/vector/riotx/features/home/room/detail/timeline/format/NoticeEventFormatter.kt:98]: You can't have both in same time, I think it's mostly in unsignedData now...

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/api/session/room/RoomService.kt [hunk 6]
```diff
@@ -104,5 +105,13 @@ interface RoomService {
                          searchOnServer: Boolean,
                          callback: MatrixCallback<Optional<String>>): Cancelable
 
+    /**
+     * Return a live data of all local changes membership who happened since the session has been opened.
```
Comment (BillCarsonFr) [inline matrix-sdk-android/src/main/java/im/vector/matrix/android/api/session/room/RoomService.kt:109]: *that happened

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/EventRelationsAggregationProcessor.kt [hunk 7]
```diff
@@ -109,7 +109,7 @@ internal class EventRelationsAggregationProcessor @Inject constructor(@UserId pr
                 return
             }
             val isLocalEcho = LocalEcho.isLocalEchoId(event.eventId ?: "")
-            when (event.getClearType()) {
+            when (event.type) {
```
Comment (BillCarsonFr) [inline matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/EventRelationsAggregationProcessor.kt:112]: I think i really wanted the clear type here.
If clear type was encrypted, i would try to see from non encrypted data if there is something i can do. 
For example if i sent a verification request, and i receive an accept that i can't decrypt I can cancel

Comment (BillCarsonFr) [reply matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/EventRelationsAggregationProcessor.kt:112]: I checked, we should keep type. It should not have been changed

File: vector/src/main/java/im/vector/riotx/features/home/room/detail/timeline/format/NoticeEventFormatter.kt (hunk 5 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/riotx/features/home/room/detail/timeline/format/NoticeEventFormatter.kt:98]: IIRC some event have both `prev_content` and `unsigned_data`, but ok anyway

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/api/session/room/RoomService.kt (hunk 6 above)
Comment (bmarty) [reply matrix-sdk-android/src/main/java/im/vector/matrix/android/api/session/room/RoomService.kt:109]: I do it

**Pull request merged**